plugins/modules/cloud/univention/udm_dns_record.py

```diff
@@ -124,14 +148,30 @@ def main():
     changed = False
     diff = None
 
+    workname = name
+    if type == 'ptr_record':
+        if not HAVE_IPADDRESS:
+            module.fail_json(msg="This module requires the 'ipaddress' python module to manage PTR records.")
+        try:
+            ipaddr_rev = ipaddress.ip_address(name).reverse_pointer
+            if zone.find('arpa') == -1:
+                raise Exception("Zone must be reversed zone for ptr_record. (e.g. 1.1.192.in-addr.arpa)")
+            subnet_offset = ipaddr_rev.find(zone)
+            if subnet_offset == -1:
+                raise Exception("IP address is not part of zone.")
```

Comment:
Inclusing the reverse address in the error message makes it easier to see what went wrong:
```suggestion
                raise Exception("IP address' reverse {0} is not part of zone".format(ipaddr_rev))
```